src/Rings/groebner.jl

```diff
@@ -162,12 +167,34 @@ function groebner_basis_with_transform(B::BiPolyArray; ord::Symbol = :degrevlex,
   return BiPolyArray(B.Ox, i), map_entries(x->B.Ox(x), m)
 end
 
-function lift_groebner_basis(I::MPolyIdeal, ord::Symbol; complete_reduction::Bool=false)
+
+@doc Markdown.doc"""
+    groebner_basis_with_transform(I::MPolyIdeal; ord::Symbol = :degrevlex, complete_reduction::Bool=false)
+
+Given an ideal `I` and optional parameters `ord` for a monomial ordering and `complete_reduction`
+this function computes a Groebner basis (if `complete_reduction = true` the reduced Groebner basis) of the
+ideal spanned by the elements in `B` w.r.t. the given monomial ordering `ord` and the
+transformation matrix from the ideal to the Groebner basis.
+
+###### Example
+```jldoctest
+julia> R,(x,y) = PolynomialRing(QQ,["x","y"])
+(Multivariate Polynomial Ring in x, y over Rational Field, fmpq_mpoly[x, y])
+
+julia> I = ideal([x*y^2-1,x^3+y^2+x*y])
+ideal generated by: x*y^2 - 1, x^3 + x*y + y^2
+
+julia> G,m = groebner_basis_with_transform(I)
+(fmpq_mpoly[x*y^2 - 1, x^3 + x*y + y^2, x^2 + y^4 + y], [1 0; 0 1; -x^2 - y y^2])
```

Comment:
Perhaps this example could be extended to illustrate the meaning of `m`? I assume something like the following works at least "in spirit" -- but I tried and it does not *actually* work like this... so as a user it'd be helpful to know how I can make it work...
```suggestion
julia> G,m = groebner_basis_with_transform(I)
(fmpq_mpoly[x*y^2 - 1, x^3 + x*y + y^2, x^2 + y^4 + y], [1 0; 0 1; -x^2 - y y^2])

julia> gens(I) * m == GB
true
```

This works but is not really "nice"
```
julia> m * AbstractAlgebra.Generic.MatSpaceElem{fmpq_mpoly}(2,1,gens(I))
[      x*y^2 - 1]
[x^3 + x*y + y^2]
[  x^2 + y^4 + y]

julia> G
3-element Array{fmpq_mpoly,1}:
 x*y^2 - 1
 x^3 + x*y + y^2
 x^2 + y^4 + y
```

Reply:
Never use MatSpaceElem - it will in general neither do nor be what you want... (MatSpaceElem{fmpz} != fmpz_mat)
`matrix(base_ring(I), ngens(I), 1, gens(I))` should work, possibly (maybe to be written) `vector(gens(I))'` as well
'`